File: Problems/Algorithms/1557. Minimum Number of Vertices to Reach All Nodes/MinVertices.kt
class Solution {
    fun findSmallestSetOfVertices(n: Int, edges: List<List<Int>>): List<Int> {
        val results: MutableList<Int> = mutableListOf()
        
        val ins = IntArray(n) { 0 }
        for (edge in edges) {
            ins[edge[1]]++
        }
        
        for (i in 0..n-1) {
            if (ins[i] == 0) results.add(i)
        }
        
        return results
    }
}

File: Problems/Algorithms/1557. Minimum Number of Vertices to Reach All Nodes/min_vertices.py
import numpy as np


class Solution:
    def findSmallestSetOfVertices(self, n: int, edges: List[List[int]]) -> List[int]:
        ins = [0 for _ in range(n)]

        for edge in edges:
            ins[edge[1]] += 1

        return np.where(np.array(ins) == 0)[0]

